File: py3/384/374.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
# def guess(num: int) -> int:

class Solution:
    def guessNumber(self, n: int) -> int:
        l, r = 1, n
        while(l < r):
            m = l + (r - l) // 2
            ans = guess(m)
            if ans == 0:
                return m
            elif ans == 1:
                l = m + 1
            else:
                r = m
        return l

File: ts/384/374.ts
/**
 * Forward declaration of guess API.
 * @param {number} num   your guess
 * @return 	            -1 if num is lower than the guess number
 *			             1 if num is higher than the guess number
 *                       otherwise return 0
 * var guess = function(num) {}
 */
function guessNumber(n: number): number {
  let l = 0, r = n
  while (l < r) {
    let m = l + ((r - l) >> 1)
    const ans = guess(m)
    if (ans === 0) return m
    else if (ans === 1) l = m + 1
    else r = m
  }
  return l
};